crypto/ml_dsa/ml_dsa_encoders.c

```diff
@@ -811,8 +813,13 @@ int ossl_ml_dsa_sk_decode(ML_DSA_KEY *key, const uint8_t *in, size_t in_len)
      * the |tr| value in the private key, else the key was corrupted.
      */
     if (!ossl_ml_dsa_key_public_from_private(key)
-            || memcmp(input_tr, key->tr, sizeof(input_tr)) != 0)
+            || memcmp(input_tr, key->tr, sizeof(input_tr)) != 0) {
+        ERR_raise_data(ERR_LIB_PROV, PROV_R_INVALID_KEY,
+                       "%s private key does not match its pubkey part",
+                       key->params->alg);
+        ossl_ml_dsa_key_reset(key);
```

Comment:
This seems fairly pointless since the callers of this function end up calling ossl_ml_dsa_free().
i.e the key failed to load. So it gets freed.

Reply:
Well, it is called from `import`, which I comment on below. I think we should keep this.